### ISSUE TITLE
Support for multiple locales directories

### ISSUE THREAD

**Issue by medihack** [felixmosh/i18next-hmr#116]
**Is your feature request related to a problem? Please describe.**
I have a relatively complex i18next setup in my Next.js app where each feature is in its own directory and has its own locales folder. I would really love to use i18next-hmr, but from looking into the documentation and source code, it seems that only a single locales folder (specified by `localesDir`) is supported.

**Describe the solution you'd like**
It would be nice to support more complex i18next setups by allowing to provide multiple locales folders, maybe by optionally allowing an array of strings for `localesDir`.

**Comment by felixmosh** [comment]
Hi @medihack, do you have multiple next.config? If so, you can configure on each one of them.
If no, I'll check if it possible to add multiple watching locale folders

**Comment by medihack** [comment]
Hi @felixmosh, thanks for the response. I only use one next.config.js. I also tried to add multiple i18next-hmr plugin instances with different `localesDir`, but this didn't work (client never refreshes then). My i18next setup is quite complex (and without next-i18next), as there are two parts on the site that can have a different active language side by side (by using two different i18n instances and custom providers). But if we could get it to work in this scenario, I guess it works everywhere. I will create a minimal example repo.

**Comment by felixmosh** [comment]
A minimal example repo will be helpful, I'll wait till then.

**Comment by medihack** [comment]
Here is a repository that demonstrates the setup with multiple locales folders: https://github.com/medihack/i18next-hmr/tree/next-complex-setup
The example is in the examples folder named `next-with-complex-setup`.

It uses a custom i18next setup that allows to have the locales side by side with its features (`app/core/locales`, `app/features/feature1/locales`, `app/features/feature2/locales`). It also maintains state after a language change (which does next-i18next not provide, at least the last time I tried it).

`i18next-hmr` is already set up, but only works for `app/core/locales` as `localesDir` (as mentioned) only allows to add one folder.

**Comment by felixmosh** [comment]
I've played with your example, I can made the plugin to support multiple locale dirs, (did it locally), but your setup is a bit weird, since you have 2 locale paths on the server but only one on the client, therefore, it won't work without a change.

The simplest change will be, 
```js
// next.config.js
const nextConfig = {
  reactStrictMode: true,
  i18n: {
    localeDetection: false,
    locales: ['de', 'en'],
    defaultLocale: 'en',
  },
  webpack(config) {
    // Setup i18next-hmr
    if (config.mode === 'development') {
      const { I18NextHMRPlugin } = require('i18next-hmr/plugin');
      const path = require('path');
      config.plugins.push(
        // TODO: Investigate how to add multiple locales folders
        new I18NextHMRPlugin({
          localesDir: [
            path.resolve(__dirname, 'app/core/locales'),
            path.resolve(__dirname, 'app/features/feature1/locales'),
            path.resolve(__dirname, 'app/features/feature2/locales'),
          ],
        })
      );
    }
    return config;
  },
};
```
Define all the locale paths for the plugin.

Move the json files from `features/feature1/locales` to `features/feature1/locales/feature1` which aligns the path with the `common` structure.
```
└── features
    ├── feature1
    │   ├── components
    │   │   └── Feature1.js
    │   └── locales
    │       └── feature1
    │           ├── de.json
    │           └── en.json
    └── feature2
        ├── components
        │   └── Feature2.js
        └── locales
            └── feature2
                ├── de.json
                └── en.json
```
If this is accepted, I will add the support for watching multiple folders.

I'm not sure what is the benefit of splitting the locales between multiple folders, i18next, introduced `namespaces` for this purpose.

There is an additional option, but I need to check if there is a bug in the lib (I get wrong lang), if so, I firstly will fix the bug, then, it should just work :]

**Comment by medihack** [comment]
Oh yes, the proposed change would be nice.  But I wonder if a modification of the Next app is really necessary, as the client fetches its translations using an [API route](https://github.com/medihack/i18next-hmr/blob/next-complex-setup/examples/next-with-complex-setup/pages/api/locales.js) which itself resolves the namespaces and both locales paths. Not sure if I overlooked something. Do you have a public branch where I could test the feature before merging it into master and publishing it?

**Comment by felixmosh** [comment]
I've pushed several improvements to language detection, as I've suspected, there was a bug, when the path contains none namespace nor language parts, such as `feature1/locales/en.json`, `feature1` resolves as a namespace which is correct, but the language remains the rest of the path, meaning, lng was `locales/en`.

Regarding your comment, your client side configured to make a fetch to `/api/locales?lng={{lng}}&namespace={{namespace}}`, the thing is that the HMR plugin should figure out from changed path (which looks like: `/Users/user/app/features/feature1/locales/en.json`) the `namespace` & the `language`, which is based on the config of `i18next`.

Anyway, the only thing that I need to add is the support for list of locale folders.

**Comment by medihack** [comment]
Sounds good. I will give it a try when it's ready and see what modifications on my side are necessary.

**Comment by felixmosh** [comment]
Check this PR, https://github.com/medihack/i18next-hmr/pull/3

**Comment by medihack** [comment]
Yes, it works great. Thanks a lot!

One question. You changed the load path of the browser client to an absolute URL. I also recognized the `[ I18NextHMR ] TypeError: Only absolute URLs are supported` on the server console only. This only happens after modifying a translation for the first time when using a relative load path URL, but it still seems to work (the translation is refreshed in the browser). What's the reasoning behind that. Do I have to use `applyServerHMR` at all?

**Comment by felixmosh** [comment]
This error is not related to i18next, but to Next.js's fetch https://github.com/node-fetch/node-fetch/issues/481,
It depends on how you configured your `i18next` backend, what happens probably is that your client only code runs on the server, Next.js runs the app on both sides.

**Comment by felixmosh** [comment]
Released in [`v1.10.0`](https://github.com/felixmosh/i18next-hmr/releases/tag/v1.10.0)

**Comment by medihack** [comment]
Yes, makes sense. Thanks a lot for your help and the lightning fast feature integration!

**Comment by felixmosh** [comment]
Thank you for using this lib 🙏🏼
